teps/0110-decouple-catalog-organization-and-reference.md

```diff
@@ -0,0 +1,235 @@
+---
+status: implementable
+title: Decouple Catalog Organization and Resource Reference
+creation-date: '2022-03-21'
+last-updated: '2022-05-26'
+authors:
+- '@vdemeester'
+- '@jerop'
+---
+
+# TEP-0110: Decouple Catalog Organization and Resource Reference
+
+<!-- toc -->
+- [Summary](#summary)
+- [Motivation](#motivation)
+  - [Goals](#goals)
+  - [Non-Goals](#non-goals)
+  - [Requirements](#requirements)
+- [Proposal](#proposal)
+  - [Tekton Bundles](#tekton-bundles)
+  - [Tekton Hub](#tekton-hub)
+    - [HTTP Endpoint](#http-endpoint)
+    - [Remote Resolution - Hub Resolver](#remote-resolution---hub-resolver)
+    - [CLI](#cli)
+  - [Tekton Catalog](#tekton-catalog)
+    - [Remote Resolution - Git Resolver](#remote-resolution---git-resolver)
+- [Future Work](#future-work)
+  - [Tekton Catalog Reorganization](#tekton-catalog-reorganization)
+  - [Tekton Catalog Support Tiers](#tekton-catalog-support-tiers)
+- [References](#references)
+<!-- /toc -->
+
+## Summary
+
+As described in [TEP-0003: Tekton Catalog Organization][tep-0003], the Tekton Catalog is a collection of blessed Tekton
+resources (`Tasks` and `Pipelines`) that can be used with any system that supports the Tekton API. On the hand, the
+[Tekton Hub][hub] to be able to search for resources.
+
+Today, versioning of resources in the Catalog is directory-based and there is tight coupling between the organization
+and how users fetch resources. Therefore, it is challenging to change the organization of the Catalog without breaking
+users.
+
+This TEP proposes to make the *Hub* and any other **official** mean of consuming Tekton resources from the community
+decoupled from where we author them. In a gist, this aims to make the repository `tektoncd/catalog` an implementation
+detail. One approach is to support long-term and non-changing URI to refer to resources via `https://` endpoints, such
```

Comment:
nit: `https://` endpoint only applies to part of the URI (OCI images do not really have that).